### ISSUE TITLE
Getting started with Wowchemy

### ISSUE THREAD

**Issue by apreshill** [rstudio/blogdown#546]
In the `NEWS.md` it says:

> When creating a new site with blogdown::new_site(), the theme gcushen/hugo-academic is automatically redirected to wowchemy/starter-academic, because the original Github repo has moved and become a repo of Hugo modules.

So `new_site("gcushen/hugo-academic")` does work, which takes care of folks stumbling upon old docs, tutorials, etc. But the following do not work, which are currently the easiest to discover via Wowchemy docs <https://wowchemy.com/docs/install/>:

`new_site(theme = "wowchemy/starter-academic")`

`new_site(theme = "wowchemy/starter-blog")`

`new_site(theme = "wowchemy/starter-book")`

`new_site(theme = "wowchemy/wowchemy-hugo-modules")`

(I realize why these don't work in theory as none are repos that contain actual Hugo themes (!!!), but not sure if any user could ever figure this out now in practice if they haven't used Hugo themes before, especially as no repository named `gcushen/hugo-academic` exists currently.)

**Comment by tcgriffith** [comment]
I suggest we just abandon wowchemy entirely. 
I object to promote a poorly supported and  commercialized theme.
risk is too high yet the gain is negligible.

**Comment by aldomann** [comment]
A couple of days I was following your [Up & Running with blogdown](https://alison.rbind.io/post/2017-06-12-up-and-running-with-blogdown/) blog post to update my Academic website to use {blogdown} and using
```r
blogdown::new_site(theme = "wowchemy/starter-academic",  theme_example = TRUE)
```

worked just fine for me.

**Comment by andrewheiss** [comment]
The issue I was running into is that blogdown installs both the template site and the theme/module thing into `themes/`, and it builds the template site simultaneously with the real site

Here's how to do it for now, which is really roundabout:

1. Run `blogdown::new_site(theme = "wowchemy/starter-academic")`
2. Blogdown will create `theme/starter-academic` (which is a template and not a theme) and `theme/github.com/wowchemy/wowchemy-hugo-modules/wowchemy` (which is the template, but packaged as a module(?))

    <img width="282" alt="image" src="https://user-images.githubusercontent.com/73663/103232953-d1e65a80-4909-11eb-8e9b-3653b2a45748.png">

3. Manually delete `theme/starter-academic` and move the deeply nested theme/module to `themes/wowchemy-hugo-modules/wowchemy`

    <img width="250" alt="image" src="https://user-images.githubusercontent.com/73663/103232970-e0347680-4909-11eb-9fbb-dfd091984816.png">


4. Change these things in `config.yaml`:

```yaml
theme: wowchemy-hugo-modules

module:
  imports:
    - path: wowchemy-hugo-modules/wowchemy
```

**Comment by aldomann** [comment]
The workaround I used to the issue @andrewheiss pointed out was to simply delete the `theme/starter-academic/content` directory, as the posts, projects, etc. were being served and deployed additionally to the local contents.

**Comment by yihui** [comment]
@apreshill If you are using the latest dev version of **blogdown**, the first three themes should just work (I've addressed the problems in 4965fe5e4ccfedcfa8af8a17968c3366a8bf7987 about a week ago). The last theme is not a theme, but pure modules, so it will never work.

@andrewheiss I've deleted the `content/` directory as @aldomann suggested, so your workaround should no longer be necessary. Thanks!

**Comment by apreshill** [comment]
@yihui reinstalled dev version and seeing the file structure looks right! But also seeing this:
![Uploading Screen Shot 2020-12-29 at 6.10.22 AM.png…]()

**Comment by apreshill** [comment]
Also, my `themes/` folder has the modules in it:
```
> dir_tree("themes/", recurse = FALSE)
themes/
├── github.com
    └── wowchemy
        └── wowchemy-hugo-modules
└── starter-academic
```

**Comment by yihui** [comment]
@apreshill That's weird. I can't reproduce it. I was aware of this problem and fixed it in 7af3de829c532e050f05c1e34a5fed1c6cb0c577. Could you verify if your `packageVersion('blogdown') >= '0.21.70'`? I assume you had restarted R after reinstalling the package.

BTW, I always use the `New Project` wizard in RStudio myself to create a new project and open it in a new R session:

![new blogdown project](https://bookdown.org/yihui/blogdown/images/new-project.png)

![new blogdown project](https://bookdown.org/yihui/blogdown/images/blogdown-project.png)

**Comment by apreshill** [comment]
Hi @yihui -

Yes I did reinstall (then restart) several times 😿 

I appear to be plagued by issues! Here is my session:

```
> install.packages("remotes")
trying URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.6/remotes_2.2.0.tgz'
Content type 'application/x-gzip' length 386257 bytes (377 KB)
==================================================
downloaded 377 KB


The downloaded binary packages are in
	/var/folders/x2/6szrwxnx4yndx0tt3ztrmdnc0000gn/T//Rtmpzqsp0e/downloaded_packages

Restarting R session...

> library(remotes)
Warning message:
package ‘remotes’ was built under R version 3.6.2 
> install_github("r-lib/remotes")
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo r-lib/remotes@HEAD
✓  checking for file ‘/private/var/folders/x2/6szrwxnx4yndx0tt3ztrmdnc0000gn/T/RtmpnAnfse/remotes10df07fb6bafb/r-lib-remotes-be48e4a/DESCRIPTION’ ...
─  preparing ‘remotes’:
✓  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘remotes_2.2.0.9000.tar.gz’
   
* installing *source* package ‘remotes’ ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (remotes)

Restarting R session...

> library(remotes)
> install_github("rstudio/blogdown")
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo rstudio/blogdown@HEAD
✓  checking for file ‘/private/var/folders/x2/6szrwxnx4yndx0tt3ztrmdnc0000gn/T/RtmpDBrwvV/remotes10e2c69c96e6a/rstudio-blogdown-4f6d023/DESCRIPTION’ ...
─  preparing ‘blogdown’:
✓  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘blogdown_0.21.70.tar.gz’
   
* installing *source* package ‘blogdown’ ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (blogdown)

Restarting R session...

> library(blogdown)
> new_site("wowchemy/starter-academic")
trying URL 'https://github.com/yihui/hugo-lithium/archive/master.tar.gz'
downloaded 109 KB

Error: module "hugo-lithium" not found; either add it as a Hugo Module or store it in "/Users/alison/rscratch/test-wow/wowchemy/starter-academic/themes".: module does not exist
Launching the server via the command:
  /Users/alison/Library/Application Support/Hugo/0.79.0/hugo server --bind 127.0.0.1 -p 4321 --themesDir themes -t hugo-lithium -D -F --navigateToChanged
Error: module "hugo-lithium" not found; either add it as a Hugo Module or store it in "/Users/alison/rscratch/test-wow/wowchemy/starter-academic/themes".: module does not exist
> xfun::session_info()
R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7, RStudio 1.4.1090

Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8

Package version:
  base64enc_0.1.3        BH_1.72.0.3           
  blogdown_0.21.70       bookdown_0.21         
  compiler_3.6.1         digest_0.6.27         
  evaluate_0.14          glue_1.4.2            
  graphics_3.6.1         grDevices_3.6.1       
  highr_0.8              htmltools_0.5.0       
  httpuv_1.5.4           jsonlite_1.7.2        
  knitr_1.30             later_1.1.0.1         
  magrittr_2.0.1         markdown_1.1          
  methods_3.6.1          mime_0.9              
  processx_3.4.4         promises_1.1.1        
  ps_1.4.0               R6_2.5.0              
  Rcpp_1.0.5             RcppTOML_0.1.6        
  rlang_0.4.9            rmarkdown_2.6         
  rstudioapi_0.13.0-9000 servr_0.21            
  stats_3.6.1            stringi_1.5.3         
  stringr_1.4.0          tinytex_0.28          
  tools_3.6.1            utils_3.6.1           
  xfun_0.19              yaml_2.2.1            
> blogdown:::serve_site()
Launching the server via the command:
  /Users/alison/Library/Application Support/Hugo/0.79.0/hugo server --bind 127.0.0.1 -p 4321 --themesDir themes -t hugo-lithium -D -F --navigateToChanged
Error: module "hugo-lithium" not found; either add it as a Hugo Module or store it in "/Users/alison/rscratch/test-wow/wowchemy/starter-academic/themes".: module does not exist
```

**Comment by apreshill** [comment]
I did a reinstall with force = TRUE and updated all packages (not those just from CRAN), and got it!

**Comment by yihui** [comment]
That's really odd. Does this file exist?

```r
file.exists('layouts/shortcodes/blogdown/postref.html')
```

Does it work after you restart R and serve the site again?

**Comment by andreashandel** [comment]
I'm late to this, but having currently the same problem of not getting Wowchemy to work.

I'm trying to start a new site using the New Project wizard as @yihui suggests above. For theme, I'm entering 'wowchemy/starter-academic'. New site seems to be generated ok. But when I try to build or serve it, I get the error message shown at bottom. 
I tried to restart R as suggested, didn't help. Also, doing the 'file.exists()` command suggested above returns FALSE.

I'm on Windows 10, using close to the latest of R/RStudio/blogdown (v 1.1) and hugo 0.79. Btw, Wowchemy+blogdown work on my own websites that I manually moved from academic to wowchemy. 

I could probably try to dig into the error message and fix, but I'm writing/updating a beginner/student tutorial and for that I just need it to work, can't ask students to get too technical. Any suggestions on how to make it work seamlessly? (Or suggestion for a nice blogdown/Hugo theme with close to the features/flexibility of Wowchemy but without the constant breaking? :) )
Thanks!

***** Error message ****
Start building sites  
ERROR 2021/03/01 12:39:01 Failed to resolve output formats: OutputFormat with key "wowchemycms_config" not found
Total in 939 ms
Error: Error building site: "C:\Users\Andreas Handel\Desktop\testsite\content\post\getting-started\index.md:50:1": failed to render shortcode "figure": failed to process shortcode: "C:\Users\Andreas Handel\Desktop\testsite\themes\github.com\wowchemy\wowchemy-hugo-modules\wowchemy\layouts\shortcodes\figure.html:17:22": execute of template failed: template: shortcodes/figure.html:17:22: executing "shortcodes/figure.html" at <resources.Get>: error calling Get: CreateFile C:\Users\Andreas Handel\Desktop\testsite\assets\media\https:\raw.githubusercontent.com\wowchemy\wowchemy-hugo-modules\master\academic.png: The filename, directory name, or volume label syntax is incorrect.
*****

**Comment by andreashandel** [comment]
Update on my previous post. I tried a slightly different approach, starting a new empty project, then initializing site with
`new_site(theme = "wowchemy/starter-academic")` as @apreshill shows it in her recent "blogdown 2021" blog entry. Same thing happens, site seems to be created right, but then when I build/serve, the same error message shows up (and again, restarting R doesn't help, but now the postref.html file is there).

**Comment by yihui** [comment]
> I'm late to this

@andreashandel You are not late---your issue is new and different. It'll be great if you could consider filing a new issue next time instead of replying under a closed issue (which we might fail to notice).

> ERROR 2021/03/01 12:39:01 Failed to resolve output formats: OutputFormat with key "wowchemycms_config" not found

That's because of https://github.com/wowchemy/starter-academic/commit/f267ba96ae17d29e21ddb201d1f8ee782af6c467#r47707726, which was a change made just a few hours ago. I'll try to automatically delete it in the dev version of **blogdown** soon. You can also delete it manually by yourself. It's in `config/_default/config.yaml`.

> Error: Error building site: "C:\Users\Andreas Handel\Desktop\testsite\content\post\getting-started\index.md:50:1": failed to render shortcode "figure":

This sounds like a bug of Hugo on Windows, since I can't reproduce it on macOS. It seems that Hugo tried to convert `/` to `\` in the figure path, which doesn't make sense in this case. You can delete the line 50 in that file. I'll try to automatically delete it, too, in the next dev version of **blogdown**. Thanks for the report!

> Or suggestion for a nice blogdown/Hugo theme with close to the features/flexibility of Wowchemy but without the constant breaking?

Our plan is the hugo-apero theme, and perhaps hugo-prose, too: https://twitter.com/xieyihui/status/1364346520649228292

**Comment by yihui** [comment]
@andreashandel Both problems should be fixed in the development version now:

```r
remotes::install_github('rstudio/blogdown')
```

**Comment by andreashandel** [comment]
@yihui Thanks for the quick reply and fix, I just tried, it works now! 

I was thinking of starting this as a new issue, but it seemed it belonged here. Will do a fresh one next time. 

And thanks for the heads up for the new hugo themes under development by you and @apreshill, I'll certainly be watching them carefully and once they are ready, likely switch my websites over.

Btw, not sure if of interest, but I needed this fix for this tutorial that I just updated: 
https://www.andreashandel.com/post/blogdown-website-1/
